src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -236,16 +257,28 @@ class WorkflowView extends Component {
                   if (value === undefined) {
                     value = config[key][name]
                   }
-                  return textInput({
-                    name, value,
-                    type: 'search',
-                    placeholder: optional ? 'Optional' : 'Required',
-                    onChange: e => {
-                      modifiedAttributes[key][name] = e.target.value
-                      this.setState({ modifiedAttributes, modified: true, untouched: false })
-                    },
-                    style: { margin: '-10px 0 -6px' }
-                  })
+
+                  const error = !optional && !value ?
+                    'This attribute is required' :
+                    this.state.invalid[key][name]
+
+                  return div({ style: { display: 'flex', alignItems: 'center', margin: '-10px -0.5rem -6px 0' } }, [
```

Comment:
This might benefit from a minimum width.

Reply:
I'm going to leave that for the overall page size thing. Asserting a min width with this table layout does dumb things.